File: Practica1/Backend-python/app/controllers/playlist_controller.py
from flask import Blueprint, request, jsonify
from app.services.playlist_service import (
    add_movie_to_playlist,
    list_user_playlist,
    delete_from_playlist
)

playlist_bp = Blueprint("playlist_bp", __name__)

# Agregar película a la playlist
@playlist_bp.route("/", methods=["POST"])
def add():
    try:
        data = request.json

        result = add_movie_to_playlist(
            data["id_usuario"],
            data["id_pelicula"]
        )

        return jsonify({
            "success": True,
            "message": "Película agregada a la lista",
            "data": result
        })

    except Exception as e:
        return jsonify({"success": False, "message": str(e)}), 400


# Obtener playlist del usuario
@playlist_bp.route("/<int:id_usuario>", methods=["GET"])
def get_user(id_usuario):
    try:
        playlist = list_user_playlist(id_usuario)

        return jsonify({
            "success": True,
            "total": len(playlist),
            "data": playlist
        })

    except Exception as e:
        return jsonify({"success": False, "message": str(e)}), 400


# Eliminar película de la playlist
@playlist_bp.route("/", methods=["DELETE"])
def remove():
    try:
        data = request.json

        delete_from_playlist(
            data["id_usuario"],
            data["id_pelicula"]
        )

        return jsonify({
            "success": True,
            "message": "Película eliminada de la lista"
        })

    except Exception as e:
        return jsonify({"success": False, "message": str(e)}), 400

File: Practica1/Backend-Nodejs/src/controllers/playlist.controller.js
const playlistService = require("../services/playlist.service");
const { formatTimeAgo } = require("../utils/time");

async function add(req, res, next) {
  try {
    const { id_usuario, id_pelicula } = req.body;

    const result = await playlistService.addMovieToPlaylist(
      id_usuario,
      id_pelicula
    );

    res.json({
      success: true,
      message: "Película agregada a tu lista 🎬",
      data: {
        id_pelicula,
        fecha_agregado: result.fecha_agregado
      }
    });

  } catch (error) {
    next(error);
  }
}

async function getUser(req, res, next) {
  try {
    const { id_usuario } = req.params;

    const playlist = await playlistService.listUserPlaylist(id_usuario);

    const bucket = process.env.AWS_BUCKET_NAME;
    const region = process.env.AWS_REGION;

    const baseUrl = `https://${bucket}.s3.${region}.amazonaws.com`;

    const data = playlist.map(p => ({
      id_pelicula: p.id_pelicula,
      titulo: p.titulo,
      poster_url: `${baseUrl}/${p.poster}`,
      director: p.director,
      anio_estreno: p.anio_estreno,
      url_contenido: p.url_contenido,
      fecha_agregado: p.fecha_agregado,
      fecha_agregado_formateada: formatTimeAgo(p.fecha_agregado)
    }));

    res.json({
      success: true,
      total: data.length,
      data
    });

  } catch (error) {
    next(error);
  }
}


async function remove(req, res, next) {
  try {
    const { id_usuario, id_pelicula } = req.body;

    await playlistService.deleteFromPlaylist(id_usuario, id_pelicula);

    res.json({
      success: true,
      message: "Película eliminada de tu lista"
    });

  } catch (error) {
    next(error);
  }
}

module.exports = {
  add,
  getUser,
  remove
};